hazelcast/src/test/java/com/hazelcast/replicatedmap/ReplicatedMapReadYourWritesTest.java

```diff
@@ -35,6 +37,31 @@
 public class ReplicatedMapReadYourWritesTest extends ReplicatedMapBaseTest {
 
     @Test
+    public void testReadYourWritesBySize() throws Exception {
+        TestHazelcastInstanceFactory nodeFactory = createHazelcastInstanceFactory();
+        HazelcastInstance instance1 = nodeFactory.newHazelcastInstance();
+        HazelcastInstance instance2 = nodeFactory.newHazelcastInstance();
+        final ReplicatedMap<Integer, Integer> map1 = instance1.getReplicatedMap("default");
+        final ReplicatedMap<Integer, Integer> map2 = instance2.getReplicatedMap("default");
+
+
+        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
+        final int count = 100;
+        for (int i = 0; i < count; i++) {
+            map.put(i, i);
+        }
+        map1.putAll(map);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                assertEquals(count, map1.size());
+                assertEquals(count, map2.size());
+            }
+        }, 5);
```

Comment:
this is a rather aggressive timeout. 

I understand you want to rule out false positives caused by anti-entropy system, but this smells like a lengthy GC could cause spurious failures. 15s or so seems better. it's an upper bound anyway, so a passing test will not slower because of this.